### ISSUE TITLE
Printable schema

### ISSUE THREAD

**Issue by chrisjlocke** [sqlitebrowser/sqlitebrowser#760]
### Details for the issue

I know the 'Database Structure' shows the database schema, and I print this out a lot when working on various databases.  I work in Windows, so its easy to do a printscreen, slap it in Word or something, and print it out.  However, one of the systems I'm working on is 17 tables, with one of these having 17 fields.  Nothing 'major', but it means printing it from the 'Database Structure' is a pain.  I use PicPick to help print scrollable windows, but it struggles on Qt widgets, so can't be used with DB4S.

In desperation, I knocked something up in VB.

![image 3](https://cloud.githubusercontent.com/assets/7872218/18174821/e0f688a4-7066-11e6-875a-f60a90efc5fe.png)

This pulls in the database schema and creates a form of treeview controls.  Its still manual, as I have to 'screenshot' it still, but PicPick is able to scroll it for me and capture it as one image.  It also fills the page left to right, rather than the 'Database Structure' view, which for me, is just one long column - not efficient with paper usage.

Requirements for this for other users might be low, so just 'throwing it out there' to see what others think, but a schema view thats printable could be quite useful.

#### I'm opening this issue because:

- [ ] DB4S is crashing
- [ ] DB4S has a bug
- [x] DB4S needs a feature
- [ ] DB4S has another problem

#### I'm using DB4S on:

- [x] Windows
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS
- [ ] Other: ___

#### I'm using DB4S version:

- [x] 3.9.0
- [ ] 3.8.0
- [ ] Other: ___

#### I have also:

- [ ] Tried out the nightly version: https://github.com/sqlitebrowser/sqlitebrowser#windows
- [ ] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Seems like a good feature to have.

@chrisjlocke So.. this might be a good starter project for your C++/Qt learning? :smile:

**Comment by chrisjlocke** [comment]
Oh crap....  :wink:

**Comment by sky5walk** [comment]
Nice, that is a good start to my earlier [schema diagram request](https://github.com/sqlitebrowser/sqlitebrowser/issues/1260).

**Comment by mgrojo** [comment]
This is, more or less, one of the points that I've implemented so far for my issue #1525. The output is very simple, but it may fit your needs. @chrisjlocke, if you'd like to try, it's already in the nightly build.

**Comment by sky5walk** [comment]
I downloaded 180927 nightly and nothing happens when I click 'Database Structure - Print'?

**Comment by chrisjlocke** [comment]
Ditto - nothing happens at all if I select 'Print' from any context menu.

**Comment by mgrojo** [comment]
Weird... Maybe we are affected by [this Qt bug](https://bugreports.qt.io/browse/QTBUG-28822?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&showAll=true) But it should be fixed in QT 5.2.1, if I understand it correctly.

**Comment by sky5walk** [comment]
Ok, can you bypass the print step and just export the canvas to an image and display it within the DB4S gui?

**Comment by mgrojo** [comment]
Let's try with a Print Preview Dialog instead of a Print Dialog. Maybe it works around the problem and nevertheless, it's nice to have it. I've changed only the Database Structure tab for the moment. Could you test it again with tomorrow's nightly build?

**Comment by mgrojo** [comment]
I've reproduced the problem under Windows. At least the print preview version works until you press the Print icon; then only the PDF export is possible, the print dialog for choosing printers, PDF format and print settings doesn't show up.

It seems that Qt has portability problems in this regard. I'll use the print preview in all the scenarios and at least we'll have PDF support.

**Comment by mgrojo** [comment]
Reading the [linked bug report](https://bugreports.qt.io/browse/QTBUG-28822?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&showAll=true):
> yes, I followed the Deployment Guide for Windows and putted the platform and printsupport folder into the folder of my executable file as well as the Qt5*.dll's

I see that we distribute a Qt5PrintSupport.dll file, but no printsupport folder. Reading [Qt for Windows - Deployment](http://doc.qt.io/qt-5/windows-deployment.html#qt-plugins)

> Your application may also depend on one or more Qt plugins, such as the print support plugin, the JPEG image format plugin or a SQL driver plugin. Be sure to distribute any Qt plugins that you need with your application. Similar to the platform plugin, each type of plugin must be located within a specific subdirectory (such as printsupport, imageformats or sqldrivers) within your distribution directory.

@justinclift, can the printsupport plugin be distributed inside the nightly build? I guess that will solve the problem. This is similar to the image-formats case that you already solved, isn't it?

**Comment by karim** [comment]
@mgrojo There's only one dll (windowsprintersupport.dll) inside **printsupport** folder. Do you mean adding this file?

**Comment by mgrojo** [comment]
@karim I suppose we have to add the complete folder with whatever it contains. If there's only that file inside, I guess that is the only one needed.

**Comment by mgrojo** [comment]
Great. That should work. Let's see with tomorrow's nightly build.

**Comment by sky5walk** [comment]
Tried latest nightly and the print action only creates a tabular listing of my database?
Was the graphical schema deployed or still in development?

**Comment by mgrojo** [comment]
No, sorry @sky5walk. I consider that another feature that it's covered in #913 and #1260. Through this development line I only plan to add icons to fields and maybe to tables, views, etc. This is just a print-out of the DB Structure view.
